lib/redis.rb

```diff
@@ -2445,7 +2445,7 @@ def pipelined
         yield(self)
         original.call_pipeline(@client)
       ensure
-        @client = original
+        @client = _client
```

Comment:
I'm not totally sure this is correct. I'll need to have a look later, but this always restore the original client, so if you nest two pipeline/multi, that might cause issues. That being said I'm not so sure such nesting is properly handled either...

Reply:
@byroot I think if you take a look at `synchronize` it references the same instance variable `@client` and passes it in as either `_client` or `client` in `pipelined` and `multi` (respectively). Given that is the same value and all the nested calls would reference the same instance value via the `synchronize` call I think it checks out. Test suite passes too which is a good sign. 

```
 def synchronize
    mon_synchronize { yield(@client) }
  end
```

Obviously could do some variable name cleanup and remove duplication but thought to keep the changeset minimal until a maintainer could take a look.

Reply:
> I think if you take a look at `synchronize`

Thanks, that's exactly the kind of thing I didn't have fresh in mind and meant to check tomorrow. So yeah,, LGTM.

> could do some variable name cleanup and remove duplication

Please do.

>  until a maintainer could take a look

One is talking with you right now 😉